mlflow/models/evaluation/default_evaluator.py

```diff
@@ -258,6 +259,38 @@ def _do_plot(**kwargs):
 }
 
 
+def _evaluate_custom_metric(custom_metric, eval_df, builtin_metrics):
+    result = custom_metric(eval_df, builtin_metrics)
+    if result is None:
+        _logger.warning(f"Custom metric function '{custom_metric.__name__}' returned None.")
```

Comment:
Not all callables have a `__name__` attribute. Can we use `getattr(custom_metric, '__name__', <default>)` here and below instead? For <default> perhaps `"at index 0"`, `"at index 1"`, ... would be helpful, as it would tell the user the index of the custom metric function in the supplied `custom_metrics` list.

https://stackoverflow.com/a/33578357/11952869